src/backend/binaryen/utils.ts

```diff
@@ -1778,31 +1779,83 @@ export namespace FunctionalFuncs {
             case ts.SyntaxKind.EqualsEqualsEqualsToken:
             case ts.SyntaxKind.ExclamationEqualsToken:
             case ts.SyntaxKind.ExclamationEqualsEqualsToken: {
-                if (rightValueType.kind === ValueTypeKind.NULL) {
-                    res = module.call(
-                        dyntype.dyntype_is_null,
-                        [dynCtx, leftValueRef],
-                        binaryen.i32,
-                    );
+                if (
+                    leftValueType.kind === ValueTypeKind.NULL ||
+                    rightValueType.kind === ValueTypeKind.NULL
+                ) {
+                    if (leftValueType.kind === ValueTypeKind.NULL) {
+                        res = module.call(
+                            dyntype.dyntype_is_null,
+                            [dynCtx, rightValueRef],
+                            binaryen.i32,
+                        );
+                    } else {
+                        res = module.call(
+                            dyntype.dyntype_is_null,
+                            [dynCtx, leftValueRef],
+                            binaryen.i32,
+                        );
+                    }
```

Comment:
These codes are highly repetitive, how about
```suggestion
                if (
                    leftValueType.kind === ValueTypeKind.NULL ||
                    rightValueType.kind === ValueTypeKind.NULL
                ) {
                    const judgedValueRef = leftValueType.kind === ValueTypeKind.NULL ?  rightValueRef : leftValueRef;
                    res = module.call(
                            dyntype.dyntype_is_null,
                            [dynCtx, judgedValueRef],
                            binaryen.i32,
                        );
                   
```

Reply:
Thanks for this perfect advice.
Done